### PR TITLE
Interface GUI and Posture Estimator

### LINKED ISSUE #58: Interface GUI with PostureEstimator
# Goal

Take the output from the PostureEstimator and use it to display the posture on the GUI

# Plan

1. Look at the PostureEstimator output and interface it to the GUI
2. Figure out how to connect the PosturePerfection and GUI projects together

### PR DESCRIPTION
**Pull request by Andrew-Ritchie**
# Details

- Combined the GUI and project Makefiles.
- Added a function to intermediate structures to convert posture status to a string.
- Integrated the posture estimator with the GUI to output determined nodes from the input image.
- Fixed linter errors within files throughout the repository.

# Screenshots/Gif (if appropriate)

Screenshot 2021-03-09 at 12 36 52

# Checklist

## Code
- [ ] I have added some tests
- [x] I have run the tests locally to ensure they all pass by running `.scripts/build.sh -enable-testing`

## Documentation
- [ ] I have updated documentation appropriately (GitHub Pages)
- [ ] I have tested the updated GitHub Pages locally to ensure the changes to documentation render correctly (see this [Guidance](https://ese-peasy.github.io/PosturePerfection/guidance.html) page)

Closes #58

### CODE REVIEW

File: src/intermediate_structures.h [hunk 1]
```diff
@@ -97,6 +98,12 @@ namespace PostProcessing {
  */
 enum Status { Untrustworthy, Trustworthy };
 
+/**
+ * @brief Converts the Untrustworthy/Trustworthy status to a string.
+ *
+ */
+std::string stringTrustworthy(Status s);
```
Comment (miklasr) [inline src/intermediate_structures.h]: Very minor, but could we rename to `stringStatus`? The enum itself is `PostProcessing::Status` and so I think that name would be a bit better.

File: src/intermediate_structures.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+std::string PostProcessing::stringTrustworthy(Status s) {
+  switch (s) {
+    case Trustworthy:
+      return "Yes";
+    case Untrustworthy:
+      return "No";
+    default:
+      return "Unknown";
+  }
+}
```
Comment (miklasr) [inline src/intermediate_structures.cpp]: Could we move this into a `PostProcessing` namespace to match the `Status` enum's namespace?

**Review by miklasr — CHANGES_REQUESTED** [review]
Look really good already. I've made some initial comments, but I haven't tried on the Pi yet. I'll get back to you once I've done that.

**Review by Ashwin-MJ — COMMENTED** [review]
Looks great! I was going to mention the same things as Miklas actually.

One other question, is do we want to try and output the results of the `PostureEstimator` on the GUI now or are we doing that in a follow up PR?

File: src/intermediate_structures.h (hunk 1 above)
Comment (Andrew-Ritchie) [reply src/intermediate_structures.h]: Thanks, no problem I will update this now.

File: src/intermediate_structures.cpp (hunk 2 above)
Comment (Andrew-Ritchie) [reply src/intermediate_structures.cpp]: Yes, no problem! :)

**Comment by Andrew-Ritchie** [comment]
Thanks @Ashwin-MJ, these values are being taken from the posture estimator but the extra data will be extracted in another PR as its related to the live feed.

**Review by miklasr — APPROVED** [review]

**Review by C-Begley — APPROVED** [review]

**Review by Ashwin-MJ — APPROVED** [review]
Looks great!

**Pull request merged**